components/base/kinematicbase/execution.go

```diff
@@ -501,3 +504,16 @@ func (ptgk *ptgBaseKinematics) makeCourseCorrectionGoals(
 	}
 	return goals
 }
+
+func (ptgk *ptgBaseKinematics) stepsToPlan(steps []arcStep, parentFrame string) motionplan.Plan {
```

Comment:
is it possible for `parentFrame` to not be world?

Reply:
Absolutely. The parent frame can come from anything that fills the Localizer interface. It could return a pose in any frame.